src/server/src/main.rs

```diff
@@ -1,23 +1,255 @@
+            {
+                user.password = update_value.into_inner(); //whatever we want update to do, just pw rn
+                return HttpResponse::Ok().json(user); 
+            }
+        }
+    }
+    HttpResponse::Unauthorized().body("Incorrect sessionId for user.")
+}
+
+async fn login_user_handler(
+    body: web::types::Json<User>,
+    path: web::types::Path<(String,)>,
+    user_map: web::types::State<UserMap>,
+    session_map: web::types::State<SessionMap>,) -> impl Responder 
+{
+    let login_user = body.into_inner();
+    let (username,) = path.into_inner();
+    let map = user_map.lock().unwrap();
+
+    if let Some(user) = map.get(&username) 
+    {
+        if user.password == login_user.password 
+        {
+            let session_id = format!("session-{}", username); //will revamp session ID generation to be better when 
+
+            let mut session_map = session_map.lock().unwrap();
+            session_map.insert(session_id.clone(), username.clone());
+
+            // 5. Return the session ID as the token to the client
+            return HttpResponse::Ok().json(&session_id);
+        }
+    }
+    HttpResponse::BadRequest().body("Invalid username/password combination.")
+}
+
+async fn logout_user_handler(
+    body: web::types::Json<String>,
+    session_map: web::types::State<SessionMap>,) -> impl Responder 
+{
+    let session_id = body.into_inner();
+    let mut session_map = session_map.lock().unwrap();
+
+    if session_map.remove(&session_id).is_some() 
+    {
+        HttpResponse::Ok().body("User successfully logged out.")
+    } else 
+    {
+        HttpResponse::BadRequest().body("Invalid sessionId.")
+    }
 }
 
 #[ntex::main]
-async fn main() -> std::io::Result<()> {
-    web::HttpServer::new(|| web::App::new().service(index).service(hello))
-        .bind(("127.0.0.1", 8080))?
-        .run()
-        .await
-}
\ No newline at end of file
+async fn main() -> std::io::Result<()> 
+{
+    Bastion::init();
+    Bastion::start();
+
+   
+    let user_map: Arc<Mutex<HashMap<String, User>>> = Arc::new(Mutex::new(HashMap::new()));
+    let session_map: Arc<Mutex<HashMap<String, ChildrenRef>>> = Arc::new(Mutex::new(HashMap::new()));
+
+    //hardcoded test user
+    {
+        let mut users = user_map.lock().unwrap();
+        users.insert(
+            "user1".to_string(),
+            User 
+            {
+                username: "user".to_string(),
+                password: "password".to_string(),
+                age: None,
+                email: None,
+                gender: None
+            },
+        );
+    }
+
+    HttpServer::new(move || 
+    {
+         // cloning arc to share it // shared ownership rust closure bs, but it doesnt error anymore
+        let user_map_clone = user_map.clone();     
+        let session_map_clone = session_map.clone();
+
+        App::new()
+            .state(user_map_clone)
+            .state(session_map_clone)
```

Comment:
turns out `.state()` can only be used one time, so you'd have to bundle the `user_map` and `session_map` into one struct and use `.state()` on that.

but, as part of the integration,`user_map` is gone. i don't have to do much

Reply:
this resolves ishty's issue with
```
I/flutter (13696): 500 <-resp status code
I/flutter (13696): App state is not configured, to configure use App::state()
```